Trying to work with tensorflow with our given json file

# 1)- Import key Modules

In [1]:
import pandas as pd
import numpy
import random

In [2]:
import nltk
from nltk.stem import WordNetLemmatizer
lemm = WordNetLemmatizer()

In [3]:
import tflearn
import tensorflow as tf

C:\Users\hassan.sherwani\Miniconda3\envs\tf1.14\lib\site-packages\tensorflow\python\framework\dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
C:\Users\hassan.sherwani\Miniconda3\envs\tf1.14\lib\site-packages\tensorflow\python\framework\dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
C:\Users\hassan.sherwani\Miniconda3\envs\tf1.14\lib\site-packages\tensorflow\python\framework\dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
C:\Users\hassan.sherwani\Miniconda3\e

curses is not supported on this machine (please install/reinstall curses for an optimal experience)








In [4]:
print(tf.__version__)

1.14.0


# 2)- Loading Data

In [5]:
data = pd.read_json('data_json.json')
data

text  \
0  helloi tried apply voucher order received mail...   
1  hi tracking number please thank youkind regard...   
2  hi ive received order one posters bad stateit ...   
3  asked test saal photobook delighted result arr...   
4     ainda recebi encomenda e estive sempre em casa   
5  dobr den mm od vs objednanou fotoknihu objednv...   

                                              intent      response  
0                                    Shipping issues  nichtkombiwb  
1                                    Shipping issues       renscom  
2  product complaints - products (Reklamation Pro...      neupverp  
3                                      ShareWithSaal       teilen2  
4                                   Order management       renscom  
5                                   Order management            wp

In [6]:
import json
with open('data_json.json') as file:
    data2 = json.load(file)

In [7]:
type(data2)

list

**Remember to check nested_json code on Github**

In [8]:
data2[2]

{'text': 'hi ive received order one posters bad stateit folded la post sent email saying reinforce packagingthe order 512005251803340regards eugenia sebastiani',
 'intent': 'product complaints - products (Reklamation Produkte)',
 'response': 'neupverp'}

#### One step deeper

In [9]:
for text in data2:
    print(text)

{'text': 'helloi tried apply voucher order received mail didnt appear order total your40 voucher 12999 rfk93r3anfpvgcan please apply orderthank colinon sat jun 6 2020 249 pm wrote http wwwsaaldigitalcom dear mr shreffler thank order pleased chosen wwwsaaldigitalcomyour order number https wwwsaaldigitalcom ordercockpit emailc40colinshrefflercomordernumber802006062223358 order article quantity total price metal print 197x295 metal print aluminium subframe 1 85 39 metal print 197x295 metal print aluminium subframe 1 85 39 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 157x197 metal print aluminium subframe 1 53 54 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 shipping costs 15 00 voucher 0 00 total 503 86 selected method

**Above is whole data**

In [10]:
type(text)

dict

### 2.1 For fetching only text from data

In [11]:
text_access=text["text"]
text_access

'dobr den mm od vs objednanou fotoknihu objednvkyhttps wwwsaaldigitalcz ordercockpit emailvvyslouzilova40seznamczordernumber512005210955444 jej pedpokldan doruen mlo bt 26 fotokniha vak stle nedorazila ke stavu objednvky nelze dohledat dn bli daje prosm proto informaci kdy bude fotokniha doruena dkuji pozdravem vyslouilov veronika'

We get only last of text data. We will append and get all text data

In [12]:
type(text_access)

str

In [13]:
word=[]

for text in data2:
    text_access=text["text"]
    word.append(text_access)

In [14]:
word

['helloi tried apply voucher order received mail didnt appear order total your40 voucher 12999 rfk93r3anfpvgcan please apply orderthank colinon sat jun 6 2020 249 pm wrote http wwwsaaldigitalcom dear mr shreffler thank order pleased chosen wwwsaaldigitalcomyour order number https wwwsaaldigitalcom ordercockpit emailc40colinshrefflercomordernumber802006062223358 order article quantity total price metal print 197x295 metal print aluminium subframe 1 85 39 metal print 197x295 metal print aluminium subframe 1 85 39 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 157x197 metal print aluminium subframe 1 53 54 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 metal print 118x177 metal print aluminium subframe 1 44 09 shipping costs 15 00 voucher 0 00 total 503 86 selected method payment

### 2.2 For all labels

In [15]:
docs_y=[]

for text in data2:
    intent_access=text["intent"]
    docs_y.append(intent_access)

In [16]:
print(docs_y)

['Shipping issues', 'Shipping issues', 'product complaints - products (Reklamation Produkte)', 'ShareWithSaal', 'Order management', 'Order management']


### 2.3 For unique labels

In [17]:
labels = []
for text in data2:
    if text['intent'] not in labels:
        labels.append(text['intent'])

In [18]:
print(labels)

['Shipping issues', 'product complaints - products (Reklamation Produkte)', 'ShareWithSaal', 'Order management']


In [19]:
print(len(docs_y)) #for all labels
print(len(labels)) # for unique labels


6
4


### 2.4 For words and unique words

In [20]:
words = []
docs_x = []


for pattern in data2: # tokenize words inside pattern 
    pattern_access=pattern['text']
    wrds = nltk.word_tokenize(pattern_access)
    words.extend(wrds)
    docs_x.append(wrds)

In [21]:
print(len(words))
print(len(docs_x))

897
6


- doc_x has 6 sentences and that is why it is of 6 length
- words has all tokens i.e 897

# 3 Preprocessing data

In [22]:
words = [lemm.lemmatize(w.lower()) for w in words if w != "?"]
words = sorted(list(set(words))) # removes any duplicates if ever occured

In [23]:
len(words)

464

In [24]:
labels = sorted(labels)
print(labels)

['Order management', 'ShareWithSaal', 'Shipping issues', 'product complaints - products (Reklamation Produkte)']


We have 4 labels. So, we will have pattern of 4 encoded terms

#  4 Bag of Words
Converting out strings of words into numerical input.

represent each sentence with a list the length of the amount of words in our models vocabulary. Each position in the list will represent a word from our vocabulary. If the position in the list is a 1 then that will mean that the word exists in our sentence, if it is a 0 then the word is nor present

bag of words because the order in which the words appear in the sentence is lost, we only know the presence of words in our models vocabulary.

In [25]:
labels

['Order management',
 'ShareWithSaal',
 'Shipping issues',
 'product complaints - products (Reklamation Produkte)']

In [26]:
training = []
output = []

out_empty = [0 for _ in range(len(labels))]

for x, doc in enumerate(docs_x):
    bag = []

    wrds = [lemm.lemmatize(w.lower()) for w in doc]

    for w in words:
        if w in wrds:
            bag.append(1)
        else:
            bag.append(0)

    output_row = out_empty[:]
    output_row[labels.index(docs_y[x])] = 1

    training.append(bag)
    output.append(output_row)

In [27]:
# Our neural network will take arrays so, we need to convert
training = numpy.array(training)
output = numpy.array(output)

In [28]:
print(training)

[[0 1 1 ... 0 0 1]
 [1 0 0 ... 0 1 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 1 0 0]]


In [29]:
len(training)

6

In [30]:
len(training[0])

464

In [31]:
#labels
print(output)

[[0 0 1 0]
 [0 0 1 0]
 [0 0 0 1]
 [0 1 0 0]
 [1 0 0 0]
 [1 0 0 0]]


In [32]:
len(labels)

4

In [33]:
len(output[0])

4

# Developing a Model

In [34]:
tf.reset_default_graph()

net = tflearn.input_data(shape=[None, len(training[0])])
net = tflearn.fully_connected(net, 8)
net = tflearn.fully_connected(net, 8)
net = tflearn.fully_connected(net, len(output[0]), activation="softmax")
net = tflearn.regression(net)

model = tflearn.DNN(net)


Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor

Instructions for updating:
keep_dims is deprecated, use keepdims instead

Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where



In [35]:
history=model.fit(training, output, n_epoch=800, batch_size=8, show_metric=True)

---------------------------------
Run id: 29KUZE
Log directory: /tmp/tflearn_logs/
INFO:tensorflow:Summary name Accuracy/ (raw) is illegal; using Accuracy/__raw_ instead.
---------------------------------
Training samples: 6
Validation samples: 0
--
Training Step: 1  | time: 0.080s
| Adam | epoch: 001 | loss: 0.00000 - acc: 0.0000 -- iter: 6/6
--
Training Step: 2  | total loss: 1.24765 | time: 0.002s
| Adam | epoch: 002 | loss: 1.24765 - acc: 0.3000 -- iter: 6/6
--
Training Step: 3  | total loss: 1.36071 | time: 0.002s
| Adam | epoch: 003 | loss: 1.36071 - acc: 0.3273 -- iter: 6/6
--
Training Step: 4  | total loss: 1.37881 | time: 0.002s
| Adam | epoch: 004 | loss: 1.37881 - acc: 0.3318 -- iter: 6/6
--
Training Step: 5  | total loss: 1.38237 | time: 0.003s
| Adam | epoch: 005 | loss: 1.38237 - acc: 0.3329 -- iter: 6/6
--
Training Step: 6  | total loss: 1.38267 | time: 0.002s
| Adam | epoch: 006 | loss: 1.38267 - acc: 0.3332 -- iter: 6/6
--
Training Step: 7  | total loss: 1.38195 | time

Training Step: 59  | total loss: 0.92746 | time: 0.001s
| Adam | epoch: 059 | loss: 0.92746 - acc: 0.6631 -- iter: 6/6
--
Training Step: 60  | total loss: 0.92180 | time: 0.003s
| Adam | epoch: 060 | loss: 0.92180 - acc: 0.6636 -- iter: 6/6
--
Training Step: 61  | total loss: 0.91642 | time: 0.002s
| Adam | epoch: 061 | loss: 0.91642 - acc: 0.6640 -- iter: 6/6
--
Training Step: 62  | total loss: 0.91127 | time: 0.002s
| Adam | epoch: 062 | loss: 0.91127 - acc: 0.6643 -- iter: 6/6
--
Training Step: 63  | total loss: 0.90633 | time: 0.001s
| Adam | epoch: 063 | loss: 0.90633 - acc: 0.6646 -- iter: 6/6
--
Training Step: 64  | total loss: 0.90157 | time: 0.002s
| Adam | epoch: 064 | loss: 0.90157 - acc: 0.6649 -- iter: 6/6
--
Training Step: 65  | total loss: 0.89698 | time: 0.002s
| Adam | epoch: 065 | loss: 0.89698 - acc: 0.6651 -- iter: 6/6
--
Training Step: 66  | total loss: 0.89253 | time: 0.002s
| Adam | epoch: 066 | loss: 0.89253 - acc: 0.6653 -- iter: 6/6
--
Training Step: 67  | tot

Training Step: 118  | total loss: 0.69713 | time: 0.002s
| Adam | epoch: 118 | loss: 0.69713 - acc: 0.6667 -- iter: 6/6
--
Training Step: 119  | total loss: 0.69353 | time: 0.002s
| Adam | epoch: 119 | loss: 0.69353 - acc: 0.6667 -- iter: 6/6
--
Training Step: 120  | total loss: 0.68994 | time: 0.002s
| Adam | epoch: 120 | loss: 0.68994 - acc: 0.6667 -- iter: 6/6
--
Training Step: 121  | total loss: 0.68635 | time: 0.002s
| Adam | epoch: 121 | loss: 0.68635 - acc: 0.6667 -- iter: 6/6
--
Training Step: 122  | total loss: 0.68276 | time: 0.002s
| Adam | epoch: 122 | loss: 0.68276 - acc: 0.6667 -- iter: 6/6
--
Training Step: 123  | total loss: 0.67918 | time: 0.002s
| Adam | epoch: 123 | loss: 0.67918 - acc: 0.6667 -- iter: 6/6
--
Training Step: 124  | total loss: 0.67560 | time: 0.002s
| Adam | epoch: 124 | loss: 0.67560 - acc: 0.6667 -- iter: 6/6
--
Training Step: 125  | total loss: 0.67202 | time: 0.002s
| Adam | epoch: 125 | loss: 0.67202 - acc: 0.6667 -- iter: 6/6
--
Training Step: 1

Training Step: 177  | total loss: 0.53168 | time: 0.002s
| Adam | epoch: 177 | loss: 0.53168 - acc: 0.7600 -- iter: 6/6
--
Training Step: 178  | total loss: 0.52775 | time: 0.002s
| Adam | epoch: 178 | loss: 0.52775 - acc: 0.7507 -- iter: 6/6
--
Training Step: 179  | total loss: 0.52394 | time: 0.002s
| Adam | epoch: 179 | loss: 0.52394 - acc: 0.7423 -- iter: 6/6
--
Training Step: 180  | total loss: 0.52026 | time: 0.002s
| Adam | epoch: 180 | loss: 0.52026 - acc: 0.7347 -- iter: 6/6
--
Training Step: 181  | total loss: 0.51667 | time: 0.002s
| Adam | epoch: 181 | loss: 0.51667 - acc: 0.7446 -- iter: 6/6
--
Training Step: 182  | total loss: 0.51319 | time: 0.002s
| Adam | epoch: 182 | loss: 0.51319 - acc: 0.7535 -- iter: 6/6
--
Training Step: 183  | total loss: 0.50979 | time: 0.002s
| Adam | epoch: 183 | loss: 0.50979 - acc: 0.7614 -- iter: 6/6
--
Training Step: 184  | total loss: 0.50647 | time: 0.002s
| Adam | epoch: 184 | loss: 0.50647 - acc: 0.7686 -- iter: 6/6
--
Training Step: 1

Training Step: 236  | total loss: 0.37550 | time: 0.002s
| Adam | epoch: 236 | loss: 0.37550 - acc: 0.9772 -- iter: 6/6
--
Training Step: 237  | total loss: 0.37270 | time: 0.002s
| Adam | epoch: 237 | loss: 0.37270 - acc: 0.9795 -- iter: 6/6
--
Training Step: 238  | total loss: 0.36983 | time: 0.002s
| Adam | epoch: 238 | loss: 0.36983 - acc: 0.9815 -- iter: 6/6
--
Training Step: 239  | total loss: 0.36687 | time: 0.002s
| Adam | epoch: 239 | loss: 0.36687 - acc: 0.9834 -- iter: 6/6
--
Training Step: 240  | total loss: 0.36383 | time: 0.002s
| Adam | epoch: 240 | loss: 0.36383 - acc: 0.9851 -- iter: 6/6
--
Training Step: 241  | total loss: 0.36069 | time: 0.002s
| Adam | epoch: 241 | loss: 0.36069 - acc: 0.9865 -- iter: 6/6
--
Training Step: 242  | total loss: 0.35746 | time: 0.001s
| Adam | epoch: 242 | loss: 0.35746 - acc: 0.9879 -- iter: 6/6
--
Training Step: 243  | total loss: 0.35413 | time: 0.002s
| Adam | epoch: 243 | loss: 0.35413 - acc: 0.9891 -- iter: 6/6
--
Training Step: 2

Training Step: 295  | total loss: 0.16953 | time: 0.001s
| Adam | epoch: 295 | loss: 0.16953 - acc: 0.9909 -- iter: 6/6
--
Training Step: 296  | total loss: 0.15797 | time: 0.001s
| Adam | epoch: 296 | loss: 0.15797 - acc: 0.9919 -- iter: 6/6
--
Training Step: 297  | total loss: 0.14734 | time: 0.001s
| Adam | epoch: 297 | loss: 0.14734 - acc: 0.9927 -- iter: 6/6
--
Training Step: 298  | total loss: 0.13756 | time: 0.002s
| Adam | epoch: 298 | loss: 0.13756 - acc: 0.9934 -- iter: 6/6
--
Training Step: 299  | total loss: 0.12854 | time: 0.003s
| Adam | epoch: 299 | loss: 0.12854 - acc: 0.9941 -- iter: 6/6
--
Training Step: 300  | total loss: 0.12023 | time: 0.002s
| Adam | epoch: 300 | loss: 0.12023 - acc: 0.9947 -- iter: 6/6
--
Training Step: 301  | total loss: 0.11257 | time: 0.001s
| Adam | epoch: 301 | loss: 0.11257 - acc: 0.9952 -- iter: 6/6
--
Training Step: 302  | total loss: 0.10550 | time: 0.002s
| Adam | epoch: 302 | loss: 0.10550 - acc: 0.9957 -- iter: 6/6
--
Training Step: 3

Training Step: 354  | total loss: 0.21319 | time: 0.001s
| Adam | epoch: 354 | loss: 0.21319 - acc: 0.9681 -- iter: 6/6
--
Training Step: 355  | total loss: 0.19378 | time: 0.002s
| Adam | epoch: 355 | loss: 0.19378 - acc: 0.9712 -- iter: 6/6
--
Training Step: 356  | total loss: 0.17633 | time: 0.001s
| Adam | epoch: 356 | loss: 0.17633 - acc: 0.9741 -- iter: 6/6
--
Training Step: 357  | total loss: 0.16064 | time: 0.001s
| Adam | epoch: 357 | loss: 0.16064 - acc: 0.9767 -- iter: 6/6
--
Training Step: 358  | total loss: 0.14651 | time: 0.001s
| Adam | epoch: 358 | loss: 0.14651 - acc: 0.9790 -- iter: 6/6
--
Training Step: 359  | total loss: 0.13379 | time: 0.001s
| Adam | epoch: 359 | loss: 0.13379 - acc: 0.9811 -- iter: 6/6
--
Training Step: 360  | total loss: 0.12231 | time: 0.002s
| Adam | epoch: 360 | loss: 0.12231 - acc: 0.9830 -- iter: 6/6
--
Training Step: 361  | total loss: 0.11196 | time: 0.001s
| Adam | epoch: 361 | loss: 0.11196 - acc: 0.9847 -- iter: 6/6
--
Training Step: 3

Training Step: 413  | total loss: 0.01010 | time: 0.001s
| Adam | epoch: 413 | loss: 0.01010 - acc: 0.9999 -- iter: 6/6
--
Training Step: 414  | total loss: 0.00995 | time: 0.001s
| Adam | epoch: 414 | loss: 0.00995 - acc: 0.9999 -- iter: 6/6
--
Training Step: 415  | total loss: 0.00981 | time: 0.003s
| Adam | epoch: 415 | loss: 0.00981 - acc: 0.9999 -- iter: 6/6
--
Training Step: 416  | total loss: 0.00967 | time: 0.002s
| Adam | epoch: 416 | loss: 0.00967 - acc: 1.0000 -- iter: 6/6
--
Training Step: 417  | total loss: 0.00954 | time: 0.002s
| Adam | epoch: 417 | loss: 0.00954 - acc: 1.0000 -- iter: 6/6
--
Training Step: 418  | total loss: 0.00941 | time: 0.002s
| Adam | epoch: 418 | loss: 0.00941 - acc: 1.0000 -- iter: 6/6
--
Training Step: 419  | total loss: 0.00928 | time: 0.002s
| Adam | epoch: 419 | loss: 0.00928 - acc: 1.0000 -- iter: 6/6
--
Training Step: 420  | total loss: 0.00916 | time: 0.002s
| Adam | epoch: 420 | loss: 0.00916 - acc: 1.0000 -- iter: 6/6
--
Training Step: 4

Training Step: 472  | total loss: 0.00555 | time: 0.001s
| Adam | epoch: 472 | loss: 0.00555 - acc: 1.0000 -- iter: 6/6
--
Training Step: 473  | total loss: 0.00551 | time: 0.001s
| Adam | epoch: 473 | loss: 0.00551 - acc: 1.0000 -- iter: 6/6
--
Training Step: 474  | total loss: 0.00547 | time: 0.002s
| Adam | epoch: 474 | loss: 0.00547 - acc: 1.0000 -- iter: 6/6
--
Training Step: 475  | total loss: 0.00543 | time: 0.001s
| Adam | epoch: 475 | loss: 0.00543 - acc: 1.0000 -- iter: 6/6
--
Training Step: 476  | total loss: 0.00538 | time: 0.002s
| Adam | epoch: 476 | loss: 0.00538 - acc: 1.0000 -- iter: 6/6
--
Training Step: 477  | total loss: 0.00534 | time: 0.002s
| Adam | epoch: 477 | loss: 0.00534 - acc: 1.0000 -- iter: 6/6
--
Training Step: 478  | total loss: 0.00530 | time: 0.001s
| Adam | epoch: 478 | loss: 0.00530 - acc: 1.0000 -- iter: 6/6
--
Training Step: 479  | total loss: 0.00526 | time: 0.001s
| Adam | epoch: 479 | loss: 0.00526 - acc: 1.0000 -- iter: 6/6
--
Training Step: 4

Training Step: 531  | total loss: 0.00370 | time: 0.002s
| Adam | epoch: 531 | loss: 0.00370 - acc: 1.0000 -- iter: 6/6
--
Training Step: 532  | total loss: 0.00368 | time: 0.002s
| Adam | epoch: 532 | loss: 0.00368 - acc: 1.0000 -- iter: 6/6
--
Training Step: 533  | total loss: 0.00366 | time: 0.002s
| Adam | epoch: 533 | loss: 0.00366 - acc: 1.0000 -- iter: 6/6
--
Training Step: 534  | total loss: 0.00364 | time: 0.002s
| Adam | epoch: 534 | loss: 0.00364 - acc: 1.0000 -- iter: 6/6
--
Training Step: 535  | total loss: 0.00361 | time: 0.001s
| Adam | epoch: 535 | loss: 0.00361 - acc: 1.0000 -- iter: 6/6
--
Training Step: 536  | total loss: 0.00359 | time: 0.002s
| Adam | epoch: 536 | loss: 0.00359 - acc: 1.0000 -- iter: 6/6
--
Training Step: 537  | total loss: 0.00357 | time: 0.002s
| Adam | epoch: 537 | loss: 0.00357 - acc: 1.0000 -- iter: 6/6
--
Training Step: 538  | total loss: 0.00355 | time: 0.002s
| Adam | epoch: 538 | loss: 0.00355 - acc: 1.0000 -- iter: 6/6
--
Training Step: 5

Training Step: 590  | total loss: 0.67569 | time: 0.001s
| Adam | epoch: 590 | loss: 0.67569 - acc: 0.9325 -- iter: 6/6
--
Training Step: 591  | total loss: 0.60842 | time: 0.001s
| Adam | epoch: 591 | loss: 0.60842 - acc: 0.9392 -- iter: 6/6
--
Training Step: 592  | total loss: 0.54789 | time: 0.001s
| Adam | epoch: 592 | loss: 0.54789 - acc: 0.9453 -- iter: 6/6
--
Training Step: 593  | total loss: 0.49343 | time: 0.001s
| Adam | epoch: 593 | loss: 0.49343 - acc: 0.9508 -- iter: 6/6
--
Training Step: 594  | total loss: 0.44443 | time: 0.001s
| Adam | epoch: 594 | loss: 0.44443 - acc: 0.9557 -- iter: 6/6
--
Training Step: 595  | total loss: 0.40036 | time: 0.001s
| Adam | epoch: 595 | loss: 0.40036 - acc: 0.9601 -- iter: 6/6
--
Training Step: 596  | total loss: 0.36071 | time: 0.001s
| Adam | epoch: 596 | loss: 0.36071 - acc: 0.9641 -- iter: 6/6
--
Training Step: 597  | total loss: 0.32505 | time: 0.001s
| Adam | epoch: 597 | loss: 0.32505 - acc: 0.9677 -- iter: 6/6
--
Training Step: 5

Training Step: 649  | total loss: 0.55260 | time: 0.002s
| Adam | epoch: 649 | loss: 0.55260 - acc: 0.9443 -- iter: 6/6
--
Training Step: 650  | total loss: 0.49814 | time: 0.002s
| Adam | epoch: 650 | loss: 0.49814 - acc: 0.9499 -- iter: 6/6
--
Training Step: 651  | total loss: 0.44914 | time: 0.002s
| Adam | epoch: 651 | loss: 0.44914 - acc: 0.9549 -- iter: 6/6
--
Training Step: 652  | total loss: 0.40505 | time: 0.002s
| Adam | epoch: 652 | loss: 0.40505 - acc: 0.9594 -- iter: 6/6
--
Training Step: 653  | total loss: 0.36537 | time: 0.002s
| Adam | epoch: 653 | loss: 0.36537 - acc: 0.9635 -- iter: 6/6
--
Training Step: 654  | total loss: 0.32966 | time: 0.002s
| Adam | epoch: 654 | loss: 0.32966 - acc: 0.9671 -- iter: 6/6
--
Training Step: 655  | total loss: 0.29751 | time: 0.001s
| Adam | epoch: 655 | loss: 0.29751 - acc: 0.9704 -- iter: 6/6
--
Training Step: 656  | total loss: 0.26856 | time: 0.002s
| Adam | epoch: 656 | loss: 0.26856 - acc: 0.9734 -- iter: 6/6
--
Training Step: 6

Training Step: 708  | total loss: 0.05149 | time: 0.002s
| Adam | epoch: 708 | loss: 0.05149 - acc: 0.9918 -- iter: 6/6
--
Training Step: 709  | total loss: 0.04734 | time: 0.002s
| Adam | epoch: 709 | loss: 0.04734 - acc: 0.9927 -- iter: 6/6
--
Training Step: 710  | total loss: 0.04361 | time: 0.002s
| Adam | epoch: 710 | loss: 0.04361 - acc: 0.9934 -- iter: 6/6
--
Training Step: 711  | total loss: 0.04025 | time: 0.002s
| Adam | epoch: 711 | loss: 0.04025 - acc: 0.9940 -- iter: 6/6
--
Training Step: 712  | total loss: 0.03723 | time: 0.002s
| Adam | epoch: 712 | loss: 0.03723 - acc: 0.9946 -- iter: 6/6
--
Training Step: 713  | total loss: 0.03450 | time: 0.002s
| Adam | epoch: 713 | loss: 0.03450 - acc: 0.9952 -- iter: 6/6
--
Training Step: 714  | total loss: 0.03204 | time: 0.002s
| Adam | epoch: 714 | loss: 0.03204 - acc: 0.9957 -- iter: 6/6
--
Training Step: 715  | total loss: 0.02982 | time: 0.002s
| Adam | epoch: 715 | loss: 0.02982 - acc: 0.9961 -- iter: 6/6
--
Training Step: 7

Training Step: 767  | total loss: 0.02893 | time: 0.002s
| Adam | epoch: 767 | loss: 0.02893 - acc: 0.9970 -- iter: 6/6
--
Training Step: 768  | total loss: 0.02688 | time: 0.002s
| Adam | epoch: 768 | loss: 0.02688 - acc: 0.9973 -- iter: 6/6
--
Training Step: 769  | total loss: 0.02502 | time: 0.002s
| Adam | epoch: 769 | loss: 0.02502 - acc: 0.9976 -- iter: 6/6
--
Training Step: 770  | total loss: 0.02334 | time: 0.002s
| Adam | epoch: 770 | loss: 0.02334 - acc: 0.9978 -- iter: 6/6
--
Training Step: 771  | total loss: 0.02181 | time: 0.002s
| Adam | epoch: 771 | loss: 0.02181 - acc: 0.9981 -- iter: 6/6
--
Training Step: 772  | total loss: 0.02042 | time: 0.001s
| Adam | epoch: 772 | loss: 0.02042 - acc: 0.9982 -- iter: 6/6
--
Training Step: 773  | total loss: 0.01916 | time: 0.002s
| Adam | epoch: 773 | loss: 0.01916 - acc: 0.9984 -- iter: 6/6
--
Training Step: 774  | total loss: 0.01801 | time: 0.001s
| Adam | epoch: 774 | loss: 0.01801 - acc: 0.9986 -- iter: 6/6
--
Training Step: 7

In [36]:
model.save("bot.tflearn")

INFO:tensorflow:C:\Users\hassan.sherwani\Documents\work\companybot_tensorflow\bot.tflearn is not in all_model_checkpoint_paths. Manually adding it.


# Predictions

with responses as per defined actions

In [37]:
def bag_of_words(s, words):
    bag = [0 for _ in range(len(words))]

    s_words = nltk.word_tokenize(s)
    s_words = [lemm.lemmatize(word.lower()) for word in s_words]

    for se in s_words:
        for i, w in enumerate(words):
            if w == se:
                bag[i] = 1
            
    return numpy.array(bag)


def chat():
    print("Start talking with the bot (type close to stop)!")
    while True:
        inp = input("User: ")
        if inp.lower() == "close":
            break

        results = model.predict([bag_of_words(inp, words)]) # shows results with probability of each neurons with that word. 
        results_index = numpy.argmax(results) # to give maximum of index out of all neurons.
        tag = labels[results_index]

        for tg in data2:
            if tg['intent'] == tag:
                responses = tg['response']

        print(random.choice(responses))

chat()

Start talking with the bot (type close to stop)!
User: hi ive received order one posters bad stateit folded la post sent email saying reinforce packagingthe order 512005251803340regards eugenia sebastiani","intent":"product complaints - products (Reklamation Produkte)","response":"neupverp"},{"text":"asked test saal photobook delighted result arrived 10 days high quality white leather look cover acrylic glass protect front photo made lovely lockdown gift best friend
2
User: close


## For checking Probability of each intent

In [38]:
def bag_of_words(s, words):
    bag = [0 for _ in range(len(words))]

    s_words = nltk.word_tokenize(s)
    s_words = [lemm.lemmatize(word.lower()) for word in s_words]

    for se in s_words:
        for i, w in enumerate(words):
            if w == se:
                bag[i] = 1
            
    return numpy.array(bag)


def chat():
    print("Start talking with the bot (type close to stop)!")
    while True:
        inp = input("User: ")
        if inp.lower() == "close":
            break

        results = model.predict([bag_of_words(inp, words)]) # shows results with probability of each neurons with that word. 
        results_index = numpy.argmax(results) # to give maximum of index out of all neurons.
        tag = labels[results_index]
        print(results)

        for tg in data2:
            if tg['intent'] == tag:
                responses = tg['response']

        print(responses)

chat()

Start talking with the bot (type close to stop)!
User: hi ive received order one posters bad stateit folded la post sent email saying reinforce packagingthe order 512005251803340regards eugenia sebastiani","intent":"product complaints - products (Reklamation Produkte)","response":"neupverp"},{"text":"asked test saal photobook delighted result arrived 10 days high quality white leather look cover acrylic glass protect front photo made lovely lockdown gift best friend
[[0.04194995 0.49493763 0.27980322 0.18330915]]
teilen2
User: ainda recebi encomenda e estive sempre em casa","intent":"Order management
[[0.9755712  0.0045671  0.00174536 0.01811626]]
wp
User: ainda recebi encomenda e estive sempre em casa","intent":"Order management
[[0.9755712  0.0045671  0.00174536 0.01811626]]
wp
User: dobr den mm od vs objednanou fotoknihu objednvkyhttps wwwsaaldigitalcz ordercockpit emailvvyslouzilova40seznamczordernumber512005210955444 jej pedpokldan doruen mlo bt 26 fotokniha vak stle nedorazila ke

# For checking with original data

In [39]:
print(labels)

['Order management', 'ShareWithSaal', 'Shipping issues', 'product complaints - products (Reklamation Produkte)']


In [40]:
print(data.intent)

0                                      Shipping issues
1                                      Shipping issues
2    product complaints - products (Reklamation Pro...
3                                        ShareWithSaal
4                                     Order management
5                                     Order management
Name: intent, dtype: object


In [41]:
print(data.response)

0    nichtkombiwb
1         renscom
2        neupverp
3         teilen2
4         renscom
5              wp
Name: response, dtype: object


In [42]:
data.head(2)

text           intent  \
0  helloi tried apply voucher order received mail...  Shipping issues   
1  hi tracking number please thank youkind regard...  Shipping issues   

       response  
0  nichtkombiwb  
1       renscom